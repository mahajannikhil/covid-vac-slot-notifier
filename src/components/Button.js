import glamorous from 'glamorous'

const Button = glamorous.button(
  {
    border: 'none',
    color: 'white',
    padding: '1em',
    borderRadius: 3,
    cursor: 'pointer'
  },
  (p, t) => ({ backgroundColor: t.colors.blue })
)

export default Button
