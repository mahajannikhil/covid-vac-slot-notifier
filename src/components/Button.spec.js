import React from 'react'
import { shallow } from 'enzyme'
import { matcher, serializer } from 'jest-glamor-react'
import Button from './Button'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('<Button />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Button theme={{ colors: { blue: 'blue' } }}>hello</Button>
    )
    expect(wrapper).toMatchSnapshotWithGlamor()
  })
})
