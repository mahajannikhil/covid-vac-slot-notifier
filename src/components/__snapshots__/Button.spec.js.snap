// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
.css-ohh20q,
[data-css-ohh20q] {
  border: none;
  color: white;
  padding: 1em;
  border-radius: 3px;
  cursor: pointer;
  background-color: blue;
}

<button
  className="css-ohh20q"
>
  hello
</button>
`;
